### ISSUE TITLE
Feature Request: Input Options For Gamepads With Fewer Buttons

### ISSUE THREAD

**Issue by ChristianSilvermoon** [Wohlstand/TheXTech#166]
In Super Mario Land 2, spin jumping was possible by jumping while holding *down*

In SMBX, pressing down mid air actually results in aerial ducking (a behaviour not normally possible in mario games) and some levels depend on the ability to do this... so I'd propose instead, an option to simulate Alt Jump when pressing Jump and Up together on the ground.

In SMB3, the Tanooki Suit's ability to turn to stone occurs specifically when pressing Run and Down... in SMBX this happens with a strange delay, seemingly when pressing Alt Run at all, sometimes seemingly requiring an additional input... so I'd propose an option to simulate Alt Run when pressing Run and tapping down very briefly, along with a release being simulated upon letting go of Run. This would allow it to work without interfering with crouch sliding.

In short, an **optional** control scheme to:
* Up + Jump (on ground) = Alt Jump
* Run + Down Tap = Alt Run until Run is released

This control scheme can cut down on the number of require buttons to play the game by two, allowing the game to be playable using an NES controller, or Raspberry Pi consoles more accurately simulating a Gameboy's button layout/count, and other devices with more limited input options.

**Comment by ds-sloth** [comment]
Awesome idea!

I can definitely add this as an option for the SDL input method, though I currently have a backlog of features to add before the abstract controls update is merged. I might prefer to get to this after the merge since it isn't absolutely essential.

How does that strike you?

**Comment by Wohlstand** [comment]
There are some gameplay options already such as changing the behavior of the Purpure Yoshi, you may utilize that config for these things too! :fox_face: :point_up:

**Comment by Alucard648** [comment]
I would better to go with Up+Run for Alt Run option.

**Comment by Wohlstand** [comment]
Also good idea, options would allow many variants for people's convenience :fox_face:

**Comment by ds-sloth** [comment]
I think this issue is out-of-scope for the current controls modules. If we want to implement it in the future, that should take place in a new minimal controls category, separate from the main SDL controller's input method type.

For now, let's consider the 10 SMBX buttons the minimum requirement for supported controllers. I think the most recent mainstream console controllers to lack this were the 3-button Mega Drive/Genesis, the Famicom/NES, and the GBC. Some PiBoys meet the requirement, and others don't.

**Comment by Wohlstand** [comment]
> Let's consider the 10 SMBX buttons the minimum requirement for supported controllers

Then I should add that to the requirements of my SMBX document.
